File: code/mapper.py
#!/usr/bin/env python
import sys
for line in sys.stdin:
    line = line.strip()
    bufs = line.split()
    for buf in bufs:
        buf = buf.strip()
        try:
            name, fru = buf.split(":") 
        except:
            continue
        name = name.strip()
        fruits = fru.split(",")
        for fruit in fruits:
            print ("%s:%s" % (fruit, name))

File: cpp/mapper.cpp
//#include <mirai>
#include <iostream>
#include <vector>
#include <string>
#include <map>
using namespace std;

void split(string str, char d, vector<string> &res){
	res.clear();
	string buf;
	for (char c:str){
		if (c == d){
			if (!buf.empty()){
				res.push_back(buf);
				buf.clear();
			}
		}else{
			buf += c;
		}
	}
	if (!buf.empty())
		res.push_back(buf);
}

int main(){
	string line;
	cin >> line;
	while (!cin.eof()){
		vector<string> nf;
		split(line, ':', nf);
		if (nf.size() < 2){
			cin >> line;
			continue;
		}
		string name = nf[0];
		vector<string> fruits;
		split(nf[1], ',', fruits);
		for (string &fruit : fruits){
			cout << fruit << ':' << name << endl;
		}
		cin >> line;
	}
	return 0;
}
